JMSC-72: Improve docs on Thread pool optimizations (#1619)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/execution-engine.adoc

appropriate processing type based on how the connector is implemented. For
details on that mechanism, refer to the xref:1.1@mule-sdk::index.adoc[Mule SDK documentation].

== Optimizations

> There are certain conditions on flow executions where the runtime may ignore the proposed Processing Type for event processor. This is due internal optimizations.

Reviewer: Is it possible to share any example conditions or a specific scenario in which this optimization happens? Proposed change: 

_Because of internal optimizations, Mule runtime engine might ignore the proposed Processing Type for some event processors. For example, thread switches are omitted when an `I/O` or `CPU_INTENSIVE` task is followed by a `CPU_LIGHT` task, because performing a thread switch in this scenario has potentially a higher impact in performance than executing the `CPU_LIGHT` task in the current thread._

Reviewer 2: I think that this is too internal, with just mentioning that this could occur is the right amount of information for a customer.

> There are certain conditions on flow executions where the runtime may ignore the proposed Processing Type for event processor. This is due internal optimizations.

Reviewer: ```suggestion
Due to internal optimizations, under certain conditions on flow executions Mule might ignore the proposed Processing Type for an event processor.
```

> There are certain conditions on flow executions where the runtime may ignore the proposed Processing Type for event processor. This is due internal optimizations. Also xref:transaction-management.adoc[Transaction Management] may affect the Proccesing Type selection.

Reviewer 2: Is confusing adding "TX Management" as part of optimization, because is not a optimization, but a change in behaviour required for the correct execution of TXs
